src/Jaeger.Core/Reporters/RemoteReporter.cs

```diff
@@ -88,25 +108,34 @@ public async Task CloseAsync(CancellationToken cancellationToken)
             }
         }
 
+        internal void FlushOnTimer(object sender, ElapsedEventArgs e)
+        {
+            try
+            {
+                Flush();
+            }
+            finally
+            {
+                _timer.Start();
+            }
+        }
+
+        internal async Task FlushAsync(CancellationToken token)
+        {
+            await Task.Run(Flush, token);
```

Comment:
You should never use `async/await` on a simple `Task`, as this is creating a two state state-machine which is only unwrapping and rewrapping the result. Just return the `Task` and remove the `async` and `await` keywords.